documentation/src/docs/security/securing-distributed-mode-cluster.md

```diff
@@ -0,0 +1,417 @@
+       - [Stage 3: Enabling TLS and Deploying Certificates](#stage-3-enabling-tls-and-deploying-certificates)
+   * [Enabling TLS and Auth in Pravega](#enabling-tls-and-auth-in-pravega)
+       - [Configuring TLS and Auth on Server Side](#configuring-tls-and-auth-on-server-side)
+       - [Configuring TLS and Credentials on Client Side](#configuring-tls-and-credentials-on-client-side)
+            * [Server Hostname Verification](#server-hostname-verification)
+       - [Having TLS and Auth Take Effect](#having-tls-and-auth-take-effect)
+   * [Conclusion](#conclusion)
+
+## Introduction
+
+In the [distributed mode](../deployment/deployment.md#pravega-modes) of running a Pravega cluster, each service runs
+separately on one or more processes, usually spread across multiple machines. The deployment options of this mode
+include:
+
+1. A manual deployment in hardware or virtual machines
+2. Containerized deployments of these types:
+    * A Kubernetes native application deployed using the Pravega Operator
+    * A Docker Compose application deployment
+    * A Docker Swarm based distributed deployment
+
+Regardless of the deployment option used, setting up Transport Layer Security (SSL/TLS) and client auth (short for
+authentication and authorization) are important steps towards a secure Pravega deployment.
+
+TLS encrypts client-server and internal communication. It also enables clients to authenticate the services running on the server nodes. Client auth enables the services to authenticate and authorize the clients. Pravega strongly recommends enabling both TLS and auth, for production clusters.
+
+Setting up security - especially TLS - in a large cluster can be daunting at first. To make it easier, this document
+provides step-by-step instructions on how to enable and configure security manually.
+
+Depending on the deployment option used and your environment, you might need to modify the steps and commands to
+suit your specific needs and policies.
+
+## Setting Up SSL/TLS
+
+There are broadly two ways of using TLS for client-server communications:
+
+1. Setup Pravega to handle TLS directly. In this case, end-to-end traffic is encrypted.
+2. Terminate TLS outside of Pravega, in an infrastructure component such as a reverse proxy or a load balancer. Traffic is encrypted until the terminating point and is in plaintext from there to Pravega.
+
+Depending on the deployment option used, it might be easier to use one or the other approach. For example, if you
+are deploying a Pravega cluster in Kubernetes, you might find approach 2 simpler and easier to manage. In a cluster deployed manually on hardware machines, it might be more convenient to use approach 1 in many cases.
+The specifics of enabling TLS will also differ depending on the deployment option used.
+
+Here, we describe the steps applicable for approach 1 in manual deployments. For approach 2, refer to the platform vendor's documentation.
```

Comment:
Would it make sense to give an example of such platform (vendor)? This sounds a bit abstract, and it would be good to make it more concrete, even if it is just for the sake of example.

Reply:
I've added a link as suggested, in the commit I just pushed. Hope that helps. We can add links for others as well later - it'll require a bit of research to point to the right references.